routes/v1.js

```diff
@@ -149,37 +228,483 @@ api.declare({
   scopes:         [
-    [
-      // Require both scopes... We need to parameterize these for each
-      // <workerType> before we return the result
-      'aws-provisioner:get-worker-type:<workerType>',
       'aws-provisioner:list-worker-types',
-    ]
   ],
   input:          undefined,  // No input
   output:         SCHEMA_PREFIX_CONST + 'list-worker-types-response.json#',
   title:          "List Worker Types",
   description: [
-    // Document what this method does
+    "List all known WorkerType names",
+  ].join('\n')
+}, function(req, res) {
+  var ctx         = this;
+  var workerType  = req.params.workerType;
+
+  if(!req.satisfies({
+    workerType:       workerType
+  })) {
+    return; // by default req.satisfies() sends a response on failure, so we're done
+  }
+
+  var p = this.WorkerType.listWorkerTypes()
+
+  p = p.then(function(workerNames) {
+    return res.reply(workerNames);
+  });
+
+  p = p.catch(function(err) {
+    errorHandler(err, res, 'listing all worker types');
+    return err;
+  });
+
+  return p;
+
+});
+
+
+api.declare({
+  method:         'get',
+  route:          '/get-launch-specs/:workerType',
+  name:           'getLaunchSpecs',
+  deferAuth:      true,
+  scopes:         ['aws-provisioner:get-worker-type:<workerType>'],
+  input:          undefined,  // No input
+  output:         SCHEMA_PREFIX_CONST + 'get-launch-specs-response.json#',
+  title:          "Get All Launch Specifications for WorkerType",
+  description: [
+    "Return the EC2 LaunchSpecifications for all combinations of regions", 
+    "and instance types or a list of reasons why the launch specifications",
+    "are not valid",
+  ].join('\n')
+}, function(req, res) {
+  var ctx         = this;
+  var workerType  = req.params.workerType;
+  var keyPrefix   = ctx.awsManager.keyPrefix
+
+  if(!req.satisfies({
+    workerType:       workerType
+  })) {
+    return; // by default req.satisfies() sends a response on failure, so we're done
+  }
+
+  var p = ctx.WorkerType.load({workerType: workerType});
+  
+  p = p.then(function(worker) {
+    return res.reply(worker.testLaunchSpecs(ctx.keyPrefix, ctx.provisionerId));
+  });
+
+  p = p.catch(function(err) {
+    errorHandler(err, res, workerType);
+  });
+
+  return p;
+
+});
+
+
+api.declare({
+  method:   'get',
+  route:    '/shutdown/every/ec2/instances/for/:workerType',
+  name:     'shutdownEveryEc2InstanceFor',
+  title:    "Shutdown Every Ec2 Instance of this Worker Type",
+  scopes:   [[
+    'aws-provisioner:aws',
+    'aws-provisioner:get-worker-type:<workerType>',
+  ]],
+  description: [
+    "WARNING: YOU ALMOST CERTAINLY DO NOT WANT TO USE THIS ",
+    "Shut down every single EC2 instance associated with this workerType. ",
+    "This means every single last one.  You probably don't want to use ",
+    "this method, which is why it has an obnoxious name.  Don't even try ",
+    "to claim you didn't know what this method does!"
```

Comment:
Ok, I've renamed it as you suggested. I've also changed it to requiring the same scopes as the turn absolutely everything off endpoint, since really, no one but super users should be doing this.